src/Plutarch/Extra/IsData.hs

```diff
@@ -72,34 +61,46 @@ import PlutusTx (
     fromData,
     toData,
  )
-import Prelude (
-    Bounded,
-    Enum,
-    Integer,
-    Maybe (Just, Nothing),
-    enumFrom,
-    error,
-    fmap,
-    foldr,
-    fromEnum,
-    fromInteger,
-    maxBound,
-    minBound,
-    pure,
-    toEnum,
-    toInteger,
-    ($),
-    (.),
-    (<$>),
- )
 
 --------------------------------------------------------------------------------
 -- ProductIsData
 
-{- |
-  Wrapper for deriving 'ToData', 'FromData' using the 'List' constructor of Data to represent a Product type.
-
-  Uses 'gProductToBuiltinData', 'gproductFromBuiltinData'.
+{- | Wrapper for deriving 'ToData', 'FromData' using the 'List'
+     constructor of Data to represent a Product type.
+
+     It is recommended to use `PlutusTypeDataList` when deriving
+     `PlutusType` as it provides some basic safety by ensuring
+     Plutarch types have an `Inner` type of `PDataRecord`.
+
+     Uses 'gProductToBuiltinData', 'gproductFromBuiltinData'.
+
+ = Example
+@
+data Foo =
+  Foo Integer [Integer]
+  deriving stock (Generic)
+  deriving anyclass (SOP.Generic)
```

Comment:
Add a qualified import to the top of the example to tell people what `SOP` refers to.